p2p/peerManager/peerManager.go

```diff
@@ -151,13 +155,21 @@ func NewManager(ctx context.Context, low int, high int, datastore datastore.Data
 		return nil, err
 	}
 
-	return &BasicPeerManager{
+	ctx, cancel := context.WithCancel(ctx)
+
+	pm := &BasicPeerManager{
 		ctx:                  ctx,
+		cancel:               cancel,
 		BasicConnMgr:         mgr,
 		BasicConnectionGater: gater,
 		streamManager:        streamManager,
 		peerDBs:              peerDBs,
-	}, nil
+		logger:               log.NewLogger("nodelogs/peers.log", "debug"),
+	}
+
+	pm.reportPeerStats()
```

Comment:
we usually just run the function as go routines instead of making a adhoc go func inside the function. i feel like that would follow what we are doing more